includes/tools/composer_package_manager.php

```diff
@@ -325,25 +334,21 @@ public function get_packages_group($write_map)
 	/**
 	 * Add a release to the packages data array.
 	 *
-	 * @param string $version Revision version.
+	 * @param string $composer_json Package composer.json content
 	 * @param int $attachment_id Attachment id for the revision download.
 	 *
 	 * @return void
 	 */
-	public function add_release($version, $attachment_id, $validated)
+	public function add_release($composer_json, $attachment_id, $validated)
 	{
-		$version .= ($validated) ? '' : '-RC';
-		$release_info = array(
-			'name'		=> $this->package_name,
-			'version'	=> $version,
-			'dist'		=> 
-				array (
-					'url'	=> titania_url::build_clean_url('download', array('id' => $attachment_id)),
-					'type'	=> 'zip',
-			),
+		$controller_helper = phpbb::$container->get('phpbb.titania.controller.helper');
+
+		$composer_json['dist'] = array(
+			'url'	=> $controller_helper->route('phpbb.titania.download', array('id' => $attachment_id)),
```

Comment:
There's a possibility of the user's sid being included in here. Use the path helper to strip that parameter.